Sources/TransferMethod/SelectTransferMethodTypePresenter.swift

```diff
@@ -251,44 +264,41 @@ final class SelectTransferMethodTypePresenter {
         }
     }
 
-    private func loadTransferMethodTypes(country: String, currency: String) {
-        sectionData.removeAll()
-        guard let result = transferMethodConfigurationKey?.transferMethodTypes(countryCode: country,
-                                                                               currencyCode: currency),
-            !result.isEmpty else {
-                view.showAlert(message: String(format: "no_transfer_method_available_error_message".localized(),
-                                               country.localized(),
-                                               currency))
-                return
+    private func loadSelectedCountry(_ countries: [HyperwalletCountry]? ) {
+        guard let countries = countries else {
+            view.showAlert(message: "no_country_available_error_message".localized())
+            return
+        }
+
+        if let userCountry = user?.country, countries.contains(where: { $0.value == userCountry }) {
+            selectedCountry = userCountry
+        } else if let country = countries.first {
+            selectedCountry = country.value
         }
-        sectionData = result
-        view.transferMethodTypeTableViewReloadData()
     }
 
-    private func loadCurrency(for country: String) {
-        guard let firstCurrency = transferMethodConfigurationKey?
-            .currencies(from: country)?.min(by: { $0.name < $1.name })
-            else {
-                view.showAlert(message: String(format: "no_currency_available_error_message".localized(),
-                                               country.localized()))
-                selectedCurrency = ""
-                view.countryCurrencyTableViewReloadData()
-                return
+    private func loadSelectedCurrency(_ currencies: [HyperwalletCurrency]?) {
+        guard let firstCurrency = currencies?.min(by: { $0.name < $1.name }) else {
+            view.showAlert(message: String(format: "no_currency_available_error_message".localized(),
+                                           selectedCountry.localized()))
+            selectedCurrency = ""
```

Comment:
showAlert will anyways go back to previous UI so this code will never be executed